### PR TITLE
implement the user profile page init functionality (#1289)

### LINKED ISSUE #1289: Implement the user profile page init functionality
In this issue, we should make sure that all the fields of the profile page work correctly and the user can interact with them easily so he/she can enter his/her data.

### PR DESCRIPTION
**Pull request by sajadkardel**
this closes #1289

### CODE REVIEW

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/EditProfile.razor.cs [hunk 1]
```diff
@@ -71,26 +94,20 @@ private async Task Save()
...
-                MessageBarText = ErrorStrings.UnknownException;
-                throw;
-            }
+            EditProfileMessage = e is KnownException 
+                ? ErrorStrings.ResourceManager.GetString(e.Message) 
+                : ErrorStrings.UnknownException;
         }
 
-        HasMessageBar = true;
+        IsLoading = false;
```
Comment (msynk) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/EditProfile.razor.cs:110]: plz use `finally` here.

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/SignUp.razor.cs [hunk 2]
```diff
@@ -32,17 +32,7 @@ public partial class SignUp
...
-            IsEnableSignUpButton = false;
-            return;
-        }
-        if (string.IsNullOrEmpty(Password))
-        {
-            IsEnableSignUpButton = false; 
-            return;
-        }
-        if (IsAcceptPrivacy is false)
+        if (string.IsNullOrEmpty(Email) || string.IsNullOrEmpty(Password) || IsAcceptPrivacy is false)
```
Comment (msynk) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/SignUp.razor.cs:35]: this change is not related to this issue, plz remove it.

**Review by msynk — APPROVED** [review]

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/EditProfile.razor.cs [hunk 3]
```diff
@@ -6,15 +6,19 @@ public partial class EditProfile
 {
     public UserDto? User { get; set; } = new();
 
+    public string? FullName { get; set; }
+    public DateTimeOffset? BirthDate { get; set; }
     public string? SelectedGender { get; set; }
```
[the remark this answers is not in the document]
Comment (sajadkardel) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/EditProfile.razor.cs:11]: We want to enable or disable the save button based on if the user input was different from what was in the database.
So we have to not change the value stored in UserDto and put user inputs in new variables so that it is finally comparable.

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/EditProfile.razor.cs [hunk 4]
```diff
@@ -52,10 +58,27 @@ protected async override Task OnInitAsync()
...
+        IsEnableSaveButton = true;
+    }
+
     private async Task Save()
     {
+        IsLoading = true;
+
         try
         {
+            User!.FullName = FullName;
```
[the remark this answers is not in the document]
Comment (sajadkardel) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/EditProfile.razor.cs:78]: You are completely right. I did not pay attention to deleting them... I did it now

**Review by msynk — APPROVED** [review]

**Review by msynk — APPROVED** [review]
